packages/babel-plugin-proposal-decorators/test/fixtures/2023-11-misc/class-in-for-loop/exec.js

```diff
@@ -51,3 +51,44 @@
 
   expect(logs.join()).toBe("0,1");
 }
+
+{
+  const logs = [];
+  const classes = [];
+  const setValueTo = i => () => () => i
+
+  for (let i = 0; i < 2; i++) {
+    classes.push(class C {
+      @setValueTo(i) [i];
+    })
+  }
+
+  for (const clazz of classes) {
+    const c = new clazz();
+    const key = Reflect.ownKeys(c)[0];
+    logs.push([key, c[key]].join(":"));
+  }
+
+  expect(logs.join()).toBe("0:0,1:1");
+}
+
+{
+  const logs = [];
+  const classes = [];
+  const noop = () => {}
+  const setValueTo = i => (_, { access, addInitializer }) => addInitializer(function() { access.set(this, i) })
+
+  for (let i = 0; i < 2; i++) {
+    classes.push(@noop class C {
+      @setValueTo(i) [i];
+    })
+  }
+
+  for (const clazz of classes) {
+    const c = new clazz();
+    const key = Reflect.ownKeys(c)[0];
+    logs.push([key, c[key]].join(":"));
+  }
+
+  expect(logs.join()).toBe("0:0,1:1");
+}
```

Comment:
For reviewers: new test cases are added to [2023-11-misc/class-in-for-loop/exec.js](https://github.com/babel/babel/pull/16331/commits/8c456e34fabc3222e116cc7e819c59c08e73ed4f#diff-ed693b1d0da32a94a2a283de6b72ee4d973fc7be055059e911e329a3d7b2015c)